modules/post/windows/manage/disable_windowsdefender_signatures.rb

```diff
@@ -24,16 +25,43 @@ def initialize(info={})
       'Platform'      => [ 'win' ],
       'SessionTypes'  => [ 'meterpreter' ],
       ))
+    register_options(
+      [
+        OptBool.new('AUTO_CLEANUP', [ true, 'Attempt to return protections after session exit', true ])
+      ])
+
   end
 
   def run
-    unless is_system?()
-      print_status("Remove Definitions Windows Defender")
-      file_path = "C:\\Program Files\\Windows Defender\\MpCmdRun.exe"
-      if exist?(file_path)
-        cmd = cmd_exec('cmd.exe', "/c \"#{file_path}\" -RemoveDefinitions -All")
+    #Are we system?
+    if not is_system?()
+      fail_with(Failure::NoAccess, "You must be System to run this Module")
+    end
+    #Is the binary there?
+    program_path = session.sys.config.getenv('ProgramFiles')
+    vprint_status("program_path = #{program_path}")
+    file_path = program_path + '\Windows Defender\MpCmdRun.exe'
+    vprint_status("file_path = #{file_path}")
+    if not exist?(file_path)
```

Comment:
I did this because I hate unless with the passion of a thousand burning suns falling into each other in the middle of a thermonuclear war during July in Texas.

Reply:
LOL

Reply:
The dispute stems from these rules in the style guide:

:eggplant:  [Use `!` instead of `not`](https://github.com/rubocop-hq/ruby-style-guide#bang-not-not)
:eggplant:  [Favor `unless` over `if` for negative conditions](https://github.com/rubocop-hq/ruby-style-guide#unless-for-negatives)

For your entertainment, you can find people continuing to argue in the various issues linked here:

hxxps://github[.]com / rubocop-hq / ruby-style-guide / issues / 329

---

:fire: :fire: :fire: :fire: :fire: 

```
# grep -rn "unless " lib/ modules/ | wc -l
4315
```

```
# grep -rn "if not " lib/ modules/ | wc -l
1616
# grep -rn "if \!" lib/ modules/ | wc -l
726
```

Reply:
Admittedly, I thought that 'not' was favored over '!', and I'm apparently in good company.... I will endeavor to favor `if !` myself in the future.   I find `if not is_system()` more readable than `if !is_system()`, but not so much I'm willing to fight it.
For `unless` I am willing to fight it.  I have seen it screw up pretty advanced Ruby developers, and even this PR used it incorrectly at first.  A lot of people who contribute are not Ruby devs, and I'm not really big on forcing an uncommon syntax on something as universally understood as `if not`.

Reply:
This is what really makes me sad....
```
git grep "unless \!" | wc -l
4
git grep "unless not " | wc -l
1
```

Reply:
I don't care much either way. Seems there's enough dispute that we shouldn't enforce one way or the other.

Personally, I find `unless` often reads better when programming defensively. `unless` also makes sense for EOL conditionals.

`unless !` and `unless not` are almost certainly wrong.